msbuild/Xamarin.Shared/Xamarin.Shared.props

```diff
@@ -40,6 +54,14 @@ Copyright (C) 2020 Microsoft. All rights reserved.
 		<!-- SdkVersion -->
 		<_SdkVersion Condition="'$(_PlatformName)' == 'macOS'">$(MacOSXSdkVersion)</_SdkVersion>
 		<_SdkVersion Condition="'$(_PlatformName)' != 'macOS'">$(MtouchSdkVersion)</_SdkVersion>
+
+		<!-- RequireCodeSigning -->
+		<_RequireCodeSigning Condition="'$(_PlatformName)' != 'macOS' And '$(_RequireCodeSigning)' == ''">false</_RequireCodeSigning> <!-- Xamarin.iOS builds are not signed by default -->
+		<_RequireCodeSigning Condition="'$(_PlatformName)' != 'macOS' And '$(ComputedPlatform)' == 'iPhone'">true</_RequireCodeSigning> <!-- except that device builds must be signed -->
+		<_RequireCodeSigning Condition="'$(_PlatformName)' == 'macOS' And '$(IsAppExtension)' == 'true'">true</_RequireCodeSigning> <!-- Xamarin.Mac extensions must be signed  -->
```

Comment:
This looks good, but is this new logic?

Reply:
No, it isn't, but my PR wasn't complete, I was missing a few things (which I've added now).

See the latest commit: c399e2466a67273c612cd6c37734cf67acb9f71a

macOS app extensions must be signed, so we have custom logic to make sure they are. The previous code would overwrite the `EnableCodeSigning` value after running `_DetectSignIdentity`, and then setting the required properties for `Codesign` to work, which turned out to be somewhat confusing. Now we're a bit more explicit, by having all the code in the same location, with comments. We're also always running `_DetectSignIdentity`, but now it knows how to not try to find a signing key if one's already been specified.